pr_parser.py

```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/env python
+
+"""
+pr_parser.py: SCRIPTS that analyse pr infomation and write parsed pr infomation
+(especially the multi-relate-pr) to post-build proprerties for downstream job.
+"""
+__author__ = "Alan Wei"
+
+
+import argparse
+import sys
+import os
+import github
+
+from github import Github
+
+class PrParser(object):
+
+    def __init__(self):
+        """
+        __repo - RackHD/on-xxxx, The repo name of which repo triggers the build 
+        __target_branch - master etc., The target branch associated with pr
+        __merge_commit_sha - origin/pr/1/merge, The merge_commit_sha associated with the pr
+        __pull_id - number in string, the pull id associated with the pr
+        __actual_commit - sha1 code, the actual commit code associated with the pr
+        """
+        self.__repo = None
+        self.__target_branch = None
+        self.__merge_commit_sha = None
+        self.__pull_id = None
+        self.__actual_commit = None
+
+    def parse_args(self, args):
+        """
+        Take in values from the user.
+        Repo, branch, merge_commit_sha and pull_id are required. This exits if they are not given.
+        :return: Parsed args for assignment
+        """
+        parser = argparse.ArgumentParser()
+        parser.add_argument("--repo",
+                            help="Git repo of the triggered pr",
+                            action="store")
+        parser.add_argument("--target_branch",
+                            help="The target branch of the pr for the named repo",
+                            action="store")
+        parser.add_argument("--actual_commit",
+                            help="The actual commit of the pr for the named repo",
+                            action="store")
+        parser.add_argument("--merge_commit_sha",
+                            help="The merge_commit_sha of target the pr",
+                            action="store")
+        parser.add_argument("--pull_id",
+                            help="The pull id of target the pr",
+                            action="store")
+        parsed_args = parser.parse_args(args)
+        return parsed_args
+
+    def assign_args(self, args):
+        """
+        Assign args to member variables.
+        :param args: Parsed args from the user
+        :return:
+        """
+        if args.repo:
```

Comment:
A little suggestion, the while function contains much similiar code, it can be more simple:
```python
for key in ["repo", "target_branch", "actual_commit", "merge_commit_sha", "pull_id"]:
    if args[key]:
        self.["__" + key] = args[key]
    else
        print "\nMust specify " + key + "\n"
    sys.exit(1)
```